### PR TITLE
[Merged by Bors] - nipost: use file instead of database to persist nipost challenge and builder state

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
part of #4090

## Changes
- save nipost challenge and builder state to a file
- remove kvstore table

### CODE REVIEW

File: activation/nipost_state.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	"os"
+	"path/filepath"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+)
+
+const (
+	challengeFilename = "NIPostChallenge"
+	builderFilename   = "NIPostBuilderState"
```
Comment (dshulyak) [inline activation/nipost_state.go]: please use better filenames (for example nipost_challenge.bin, nipost_builder_state.bin)

File: activation/nipost_state.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+func saveNipostChallenge(dir string, ch *types.NIPostChallenge) error {
+	if _, err := os.Stat(dir); err != nil {
+		return fmt.Errorf("post dir not created: %w", err)
+	}
+	data, err := codec.Encode(ch)
+	if err != nil {
+		return fmt.Errorf("encode nipost challenge: %w", err)
+	}
+	err = os.WriteFile(filepath.Join(dir, challengeFilename), data, ownerReadWrite)
```
Comment (dshulyak) [inline activation/nipost_state.go]: as this data is somewhat critical i think we should report success only after fsync and also write checksum that is checked on the load side

write without fsync is not guaranteed to land on disk. or may land partially, in this case there will be undefined behavior on load

Comment (countvonzero) [reply activation/nipost_state.go]: thanks. done

**Comment by countvonzero** [comment]
bors try

**Comment by countvonzero** [comment]
bors try

**Review by dshulyak — APPROVED** [review]

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**